agents/abstraction_agent.py

```diff
@@ -39,10 +41,7 @@ def __init__(
 
         self.project_name = project_name
         self.meta_context = meta_context
-
-        self.context: dict[str, AnalysisInsights | list[AnalysisInsights]] = {
-            "structure_insight": []
-        }  # Store evolving insights here
+        self.context: dict[str, LLMBaseModel] = {"structure_insight": []}
```

Comment:
This might be the reason why you can't automatically merge, I removed the last part because @ivanmilevtues mentioned we don't need it.
```suggestion
        self.context: dict[str, LLMBaseModel] = {}
```